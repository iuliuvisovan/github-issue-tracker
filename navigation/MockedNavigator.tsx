import React, { FunctionComponent } from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

const Stack = createStackNavigator();
const MockedNavigator = ({ component, params = {} }: { component: FunctionComponent<any>; params?: object }) => {
  return (
    <NavigationContainer>
      <Stack.Navigator>
        <Stack.Screen name="MockedScreen" component={component} initialParams={params} />
      </Stack.Navigator>
    </NavigationContainer>
  );
};

export default MockedNavigator;
