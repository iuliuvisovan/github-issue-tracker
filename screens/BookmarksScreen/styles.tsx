import { Color } from '../../components';
import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    paddingHorizontal: 16,
    backgroundColor: Color.pageBackground,
  },
});
