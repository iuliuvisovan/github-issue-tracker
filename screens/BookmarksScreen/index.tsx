import React, { useEffect } from 'react';

import { ActivityIndicator, View } from 'react-native';
import { FlatList } from 'react-native-gesture-handler';

import { useDispatch, useSelector } from 'react-redux';
import { IApplicationState } from '../../redux';
import * as bookmarkActions from '../../redux/actions/bookmarks';

import { Color } from '../../components';
import Issue from '../IssuesScreen/Issue';
import styles from './styles';
import { IBookmarksScreenProps } from '../../types/navigation';

export default function BookmarksScreen(props: IBookmarksScreenProps) {
  const { bookmarks: bookmarks, loading } = useSelector((state: IApplicationState) => state.bookmarksReducer);

  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(bookmarkActions.getBookmarks());
  }, []);

  return (
    <View style={styles.container}>
      {loading && <ActivityIndicator color={Color.blue} style={{ marginTop: 16 }} />}
      <FlatList
        data={bookmarks}
        contentContainerStyle={{ paddingTop: 16 }}
        keyExtractor={(item) => item.id + ''}
        renderItem={({ item }) => <Issue issue={{ ...item, isBookmarked: true }} navigation={props.navigation} />}
      />
    </View>
  );
}
