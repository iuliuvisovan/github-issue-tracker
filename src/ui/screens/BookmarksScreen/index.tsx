import React, { useEffect } from 'react';
import { ActivityIndicator, View, FlatList } from 'react-native';
import { Feather } from '@expo/vector-icons';

import { Color, Text } from '../../components';
import useBookmarks from '../../hooks/useBookmarks';
import styles from './styles';

import Issue from '../IssuesScreen/Issue';
import { BookmarksScreenProps } from '../../../data/types/navigation';

export default function BookmarksScreen(props: BookmarksScreenProps) {
  const bookmarksManager = useBookmarks();
  const { bookmarks, loading } = bookmarksManager.data;
  const { getBookmarks } = bookmarksManager.actions;

  useEffect(getBookmarks, []);

  return (
    <View style={styles.container}>
      {loading && <ActivityIndicator color={Color.blue} style={{ marginTop: 16 }} />}
      {!bookmarks.length && (
        <View style={styles.noBookmarks}>
          <Feather size={60} color={Color.border + '88'} name="star" style={{ marginRight: 4 }} />
          <Text style={styles.noBookmarksText}>Bookmarked issues will appear here!</Text>
        </View>
      )}
      <FlatList
        data={bookmarks}
        contentContainerStyle={{ paddingTop: 16 }}
        keyExtractor={(item) => item.id + ''}
        renderItem={({ item }) => <Issue issue={{ ...item, isBookmarked: true }} navigation={props.navigation} />}
      />
    </View>
  );
}
