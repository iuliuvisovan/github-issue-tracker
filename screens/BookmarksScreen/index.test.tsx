import React from 'react';
import BookmarksScreen from './index';
import MockedNavigator from '../../navigation/MockedNavigator';
import { create, act } from 'react-test-renderer';
import { Provider as ReduxProvider } from 'react-redux';
import store from '../../redux';

test('snapshot', async () => {
  let tree;

  await act(async () => {
    tree = create(
      <ReduxProvider store={store}>
        <MockedNavigator component={BookmarksScreen} />
      </ReduxProvider>
    );
  });

  expect(tree).toMatchSnapshot();
});
