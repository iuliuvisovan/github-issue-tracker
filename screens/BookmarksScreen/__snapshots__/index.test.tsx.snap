// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "zIndex": 1,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                aria-level="1"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "#f4f6f6",
                          "flex": 1,
                          "paddingHorizontal": 16,
                        }
                      }
                    >
                      <RCTScrollView
                        contentContainerStyle={
                          Object {
                            "paddingTop": 16,
                          }
                        }
                        data={
                          Array [
                            Object {
                              "body": "mock body",
                              "id": 1,
                              "isBookmarked": true,
                              "labels": Array [
                                Object {
                                  "color": "#000",
                                  "default": false,
                                  "description": "mock description",
                                  "id": "2",
                                  "name": "mock label",
                                  "url": "mock_url",
                                },
                              ],
                              "state": "open",
                              "title": "Mock",
                              "user": Object {
                                "avatar_url": "mock_url",
                                "html_url": "mock_url",
                                "id": 1,
                                "login": "mock_login",
                                "url": "mock_url",
                              },
                            },
                            Object {
                              "body": "mock body",
                              "id": 2,
                              "isBookmarked": true,
                              "labels": Array [
                                Object {
                                  "color": "#000",
                                  "default": false,
                                  "description": "mock description",
                                  "id": "2",
                                  "name": "mock label",
                                  "url": "mock_url",
                                },
                              ],
                              "state": "open",
                              "title": "Mock",
                              "user": Object {
                                "avatar_url": "mock_url",
                                "html_url": "mock_url",
                                "id": 2,
                                "login": "mock_login",
                                "url": "mock_url",
                              },
                            },
                          ]
                        }
                        disableVirtualization={false}
                        getItem={[Function]}
                        getItemCount={[Function]}
                        horizontal={false}
                        initialNumToRender={10}
                        keyExtractor={[Function]}
                        maxToRenderPerBatch={10}
                        onContentSizeChange={[Function]}
                        onEndReachedThreshold={2}
                        onLayout={[Function]}
                        onMomentumScrollEnd={[Function]}
                        onScroll={[Function]}
                        onScrollBeginDrag={[Function]}
                        onScrollEndDrag={[Function]}
                        removeClippedSubviews={false}
                        renderItem={[Function]}
                        scrollEventThrottle={50}
                        stickyHeaderIndices={Array []}
                        updateCellsBatchingPeriod={50}
                        viewabilityConfigCallbackPairs={Array []}
                        windowSize={21}
                      >
                        <View>
                          <View
                            onLayout={[Function]}
                            style={null}
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "backgroundColor": "#fff",
                                  "borderRadius": 8,
                                  "elevation": 4,
                                  "marginBottom": 16,
                                  "opacity": 1,
                                  "paddingHorizontal": 16,
                                  "paddingVertical": 16,
                                  "shadowColor": "#414d5b",
                                  "shadowOffset": Object {
                                    "height": 2,
                                    "width": 2,
                                  },
                                  "shadowOpacity": 0.1,
                                  "shadowRadius": 3,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "marginBottom": 16,
                                    "marginTop": -2,
                                  }
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "uri": "mock_url",
                                    }
                                  }
                                  style={
                                    Object {
                                      "borderRadius": 10,
                                      "height": 20,
                                      "marginRight": 8,
                                      "marginTop": 1,
                                      "width": 20,
                                    }
                                  }
                                />
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#14181c",
                                        "fontFamily": "muli",
                                        "lineHeight": 16,
                                      },
                                      Object {
                                        "color": "#414d5b",
                                        "fontFamily": "muli-semi-bold",
                                        "fontSize": 12,
                                      },
                                    ]
                                  }
                                >
                                  mock_login
                                </Text>
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#14181c",
                                        "fontFamily": "muli",
                                        "lineHeight": 16,
                                      },
                                      Object {
                                        "color": "#aab2ba",
                                        "fontSize": 12,
                                      },
                                    ]
                                  }
                                >
                                   
                                  • 
                                  a few seconds ago
                                </Text>
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#14181c",
                                        "fontFamily": "muli",
                                        "lineHeight": 16,
                                      },
                                      Object {
                                        "color": "#aab2ba",
                                        "fontSize": 12,
                                      },
                                    ]
                                  }
                                >
                                   • 
                                </Text>
                                <View
                                  style={Object {}}
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#aab2ba",
                                          "fontSize": 16,
                                        },
                                        undefined,
                                        Object {
                                          "fontFamily": "anticon",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        Object {},
                                      ]
                                    }
                                  >
                                    
                                  </Text>
                                </View>
                              </View>
                              <View
                                style={
                                  Array [
                                    Object {
                                      "borderBottomLeftRadius": 16,
                                      "borderTopLeftRadius": 16,
                                      "paddingLeft": 12,
                                      "paddingRight": 14,
                                      "paddingVertical": 4,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 13,
                                    },
                                    Object {
                                      "backgroundColor": "#16c65d",
                                    },
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#14181c",
                                        "fontFamily": "muli",
                                        "lineHeight": 16,
                                      },
                                      Object {
                                        "color": "#fff",
                                        "fontFamily": "muli-extra-bold",
                                      },
                                    ]
                                  }
                                >
                                  Open
                                </Text>
                              </View>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#14181c",
                                      "fontFamily": "muli",
                                      "lineHeight": 16,
                                    },
                                    Object {
                                      "maxWidth": 734,
                                    },
                                  ]
                                }
                              >
                                Mock
                              </Text>
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                    "flexWrap": "wrap",
                                    "marginTop": 8,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "borderRadius": 16,
                                        "flexDirection": "row",
                                        "height": 22,
                                        "marginRight": 8,
                                        "marginTop": 8,
                                        "paddingHorizontal": 10,
                                      },
                                      Object {
                                        "backgroundColor": "##000",
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "color": "#14181c",
                                          "fontFamily": "muli",
                                          "lineHeight": 16,
                                        },
                                        Object {
                                          "fontFamily": "muli-semi-bold",
                                          "fontSize": 12,
                                        },
                                        Object {
                                          "color": "#fff",
                                        },
                                      ]
                                    }
                                  >
                                    mock label
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            onLayout={[Function]}
                            style={null}
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "backgroundColor": "#fff",
                                  "borderRadius": 8,
                                  "elevation": 4,
                                  "marginBottom": 16,
                                  "opacity": 1,
                                  "paddingHorizontal": 16,
                                  "paddingVertical": 16,
                                  "shadowColor": "#414d5b",
                                  "shadowOffset": Object {
                                    "height": 2,
                                    "width": 2,
                                  },
                                  "shadowOpacity": 0.1,
                                  "shadowRadius": 3,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "alignItems": "center",
                                    "flexDirection": "row",
                                    "marginBottom": 16,
                                    "marginTop": -2,
                                  }
                                }
                              >
                                <Image
                                  source={
                                    Object {
                                      "uri": "mock_url",
                                    }
                                  }
                                  style={
                                    Object {
                                      "borderRadius": 10,
                                      "height": 20,
                                      "marginRight": 8,
                                      "marginTop": 1,
                                      "width": 20,
                                    }
                                  }
                                />
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#14181c",
                                        "fontFamily": "muli",
                                        "lineHeight": 16,
                                      },
                                      Object {
                                        "color": "#414d5b",
                                        "fontFamily": "muli-semi-bold",
                                        "fontSize": 12,
                                      },
                                    ]
                                  }
                                >
                                  mock_login
                                </Text>
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#14181c",
                                        "fontFamily": "muli",
                                        "lineHeight": 16,
                                      },
                                      Object {
                                        "color": "#aab2ba",
                                        "fontSize": 12,
                                      },
                                    ]
                                  }
                                >
                                   
                                  • 
                                  a few seconds ago
                                </Text>
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#14181c",
                                        "fontFamily": "muli",
                                        "lineHeight": 16,
                                      },
                                      Object {
                                        "color": "#aab2ba",
                                        "fontSize": 12,
                                      },
                                    ]
                                  }
                                >
                                   • 
                                </Text>
                                <View
                                  style={Object {}}
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#aab2ba",
                                          "fontSize": 16,
                                        },
                                        undefined,
                                        Object {
                                          "fontFamily": "anticon",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        Object {},
                                      ]
                                    }
                                  >
                                    
                                  </Text>
                                </View>
                              </View>
                              <View
                                style={
                                  Array [
                                    Object {
                                      "borderBottomLeftRadius": 16,
                                      "borderTopLeftRadius": 16,
                                      "paddingLeft": 12,
                                      "paddingRight": 14,
                                      "paddingVertical": 4,
                                      "position": "absolute",
                                      "right": 0,
                                      "top": 13,
                                    },
                                    Object {
                                      "backgroundColor": "#16c65d",
                                    },
                                  ]
                                }
                              >
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#14181c",
                                        "fontFamily": "muli",
                                        "lineHeight": 16,
                                      },
                                      Object {
                                        "color": "#fff",
                                        "fontFamily": "muli-extra-bold",
                                      },
                                    ]
                                  }
                                >
                                  Open
                                </Text>
                              </View>
                              <Text
                                style={
                                  Array [
                                    Object {
                                      "color": "#14181c",
                                      "fontFamily": "muli",
                                      "lineHeight": 16,
                                    },
                                    Object {
                                      "maxWidth": 734,
                                    },
                                  ]
                                }
                              >
                                Mock
                              </Text>
                              <View
                                style={
                                  Object {
                                    "flexDirection": "row",
                                    "flexWrap": "wrap",
                                    "marginTop": 8,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "alignItems": "center",
                                        "borderRadius": 16,
                                        "flexDirection": "row",
                                        "height": 22,
                                        "marginRight": 8,
                                        "marginTop": 8,
                                        "paddingHorizontal": 10,
                                      },
                                      Object {
                                        "backgroundColor": "##000",
                                      },
                                    ]
                                  }
                                >
                                  <Text
                                    style={
                                      Array [
                                        Object {
                                          "color": "#14181c",
                                          "fontFamily": "muli",
                                          "lineHeight": 16,
                                        },
                                        Object {
                                          "fontFamily": "muli-semi-bold",
                                          "fontSize": 12,
                                        },
                                        Object {
                                          "color": "#fff",
                                        },
                                      ]
                                    }
                                  >
                                    mock label
                                  </Text>
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Object {
                                "height": 0,
                              }
                            }
                          />
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
